jetty9/src/com/thoughtworks/go/server/Jetty9Server.java

```diff
@@ -109,11 +109,13 @@ public void addExtraJarsToClasspath(String extraClasspath) {
     }
 
     @Override
-    public void setSessionAndCookieExpiryTimeout(int sessionAndCookieExpiryTimeout) {
-        SessionCookieConfig cookieConfig = webAppContext.getSessionHandler().getSessionManager().getSessionCookieConfig();
-        cookieConfig.setHttpOnly(true);
-        cookieConfig.setMaxAge(sessionAndCookieExpiryTimeout);
+    public void setSessionCookieConfig() {
+        int sessionAndCookieExpiryTimeout = systemEnvironment.sessionTimeoutInSeconds();
         SessionManager sessionManager = webAppContext.getSessionHandler().getSessionManager();
+        SessionCookieConfig sessionCookieConfig = sessionManager.getSessionCookieConfig();
+        sessionCookieConfig.setHttpOnly(true);
+        sessionCookieConfig.setSecure(systemEnvironment.isSessionCookieSecure());
+        sessionCookieConfig.setMaxAge(sessionAndCookieExpiryTimeout);
```

Comment:
Did we talk about making this configurable, so that it can be set to `0`, so that it's a proper session cookie? That is, it's wiped out on browser close?

Reply:
There were some discussions around it a few times but I get a feeling we might have missed talking to you about it. The value for max-age was hardcoded to 2 weeks, but as part of 17.11 this was made configurable to allow users to reduce this to to a shorter period, that could mean setting it to `-1` to make it a proper session cookie, or to set it to say 1 hr(`3600`) or so if someone wanted it to be that way. This PR allows configuring it to be marked as secure. It would work if the installation was accessible via https only, else it would disallow users from logging in when using http with the flag set. The default behavior for both of these are the same as they used to be, intentionally.  Any concerns?

Reply:
Sounds good, thanks. Is this documented?

Reply:
https://docs.gocd.org/current/configuration/dev_authentication.html 
Look for `go.server.session.timeout.seconds`, doesn't mention about `-1` - may be we should. Yet to add a note about the secure flag.

Reply:
Ok!

Reply:
@arvindsv - Thanks for probing. We found an issue with the current implementation. Submitted a PR for the same -> https://github.com/gocd/gocd/pull/4099
Submitted a PR for the docs too -> https://github.com/gocd/docs.go.cd/pull/133